src/test/java/com/agnieszkapawska/flashcards/api/controllers/FlashcardControllerTest.java

```diff
@@ -0,0 +1,40 @@
+package com.agnieszkapawska.flashcards.api.controllers;
+
+import com.agnieszkapawska.flashcards.FlashcardsApplicationAbstractTests;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardDto;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveResponseDto;
+import com.agnieszkapawska.flashcards.domain.exceptions.EntityNotCreatedException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class FlashcardControllerTest extends FlashcardsApplicationAbstractTests {
+
+    @Test
+    public void shouldReturnStatusOkWhenSavingFlashcard() {
+        //given
+        when(super.flashcardFacade.saveFlashcard(any(FlashcardDto.class))).thenReturn(super.flashcardSaveResponseDto);
+        //when
+        ResponseEntity<FlashcardSaveResponseDto> responseEntity =
+                testRestTemplate.postForEntity(super.baseUrl + "/flashcard", super.flashcardDto, FlashcardSaveResponseDto.class);
+        //then
+        Assert.assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+    }
+
+    @Test
+    public void shouldReturnStatusConflictWhenSavingFlashcard() {
```

Comment:
```suggestion
    public void should_ReturnStatusConflict_WhenSavingFlashcard_WhenCouldNotCreateEntity() {
```
Test names should help to understand what happens in test. Now we have two tests for saving flashcard and one returns status conflict and other returns status OK. And we need to dive into test details to find out what and when happens. 
Also I did propose to add `_` - it makes more readable names. We can clearly see separated parts (what should happen, when should happen) in name.